pkgs/tools/misc/findup/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, stdenv, fetchFromGitHub, zig, testVersion, findup }:
+
+stdenv.mkDerivation rec {
+  pname = "findup";
+  version = "1.0";
+
+  src = fetchFromGitHub {
+    owner = "hiljusti";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-erlKIiYYlWnhoeD3FnKdxnHjfGmmJVXk44DUja5Unig=";
+  };
+
+  nativeBuildInputs = [ zig ];
+
+  # Builds and installs (at the same time) with Zig.
+  dontConfigure = true;
+  dontBuild = true;
+
+  # Give Zig a directory for intermediate work.
+  preInstall = ''
+    export HOME=$TMPDIR
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    zig build -Drelease-safe -Dcpu=baseline --prefix $out
+    runHook postInstall
+  '';
+
+  passthru.tests.version = testVersion { package = findup; };
+
+  meta = with lib; {
+    homepage = "https://github.com/hiljusti/findup";
+    description = "Search parent directories for sentinel files";
+    license = licenses.mit;
+    maintainers = with maintainers; [ hiljusti ];
+
+    # Broken until Nix supports newer Mac, or Zig supports older Mac.
+    # Nix supports macOS up to 10.12: https://github.com/NixOS/nixpkgs/issues/86299
+    # Zig supports macOS 10.13 or newer: https://github.com/ziglang/zig/issues/5023#issuecomment-612763356
+    broken = stdenv.hostPlatform.isDarwin;
```

Comment:
If this broken is only because of Zig compiler then you should leave it without. Nix is clevar enough to set broken for packages that require zig.

Reply:
Ahh thanks, I didn't know!